### DISCUSSION TITLE
Allow Analog Pin Usage

### DISCUSSION THREAD

**Discussion by ftrocchi** [DeanIsMe/SevSeg#92] [Ideas]
As we all know, the analog pins can be used as digital pins.  Why not create a constructor that allows a mix of pins (analog or digital) for both the digit pins and segment pins?  

My use case:  I have an Arduino Nano and a 3 digit 7 segment (37S) display.  The 37S pins will fit nicely on top of the Nano, but some of the pins are on the Analog Pins.  For instance I would like to use pins D7-D2 (6 pins) on the top and A2-A6 (5 pins) on the bottom of the Nano.

Thanks for your time.

**Comment by machmar** [comment]
I am pretty sure you can just write the name of the analog pin instead of the regular pin number.
Please close this issue.

edit: sorry for the "Please close this issue". This is not an issue, my bad.

**Reply by ftrocchi** [reply]
I tried earlier today.   You can't put 'A2' in as a pin because it doesn't translate into a specific BYTE.  The constructor takes a byte array, so therefore no analog pins work at the moment.

**Reply by ftrocchi** [reply]
I just read that there are numbers for A0-A7 (14... ).  So let me try that.  I may just have been dumb :)

**Reply by machmar** [reply]
A2 is a macro - it has to work

`#define A0 14`
the compiler replaces A0 with the number 14

**Reply by ftrocchi** [reply]
Yeah, I was an idiot.  lol.  thanks for letting me talk it through.  This idea can go away :)

**Reply by machmar** [reply]
lol no problem
